pdf-workitem/src/main/java/org/jbpm/process/workitem/pdf/GeneratePDFWorkitemHandler.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.process.workitem.pdf;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.jbpm.document.Document;
+import org.jbpm.process.workitem.core.AbstractLogOrThrowWorkItemHandler;
+import org.jbpm.process.workitem.core.util.RequiredParameterValidator;
+import org.jbpm.process.workitem.core.util.Wid;
+import org.jbpm.process.workitem.core.util.WidMavenDepends;
+import org.jbpm.process.workitem.core.util.WidParameter;
+import org.jbpm.process.workitem.core.util.WidResult;
+import org.jbpm.process.workitem.core.util.service.WidAction;
+import org.jbpm.process.workitem.core.util.service.WidService;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.thymeleaf.TemplateEngine;
+import org.thymeleaf.context.Context;
+import org.thymeleaf.templateresolver.StringTemplateResolver;
+import org.w3c.tidy.Tidy;
+import org.xhtmlrenderer.pdf.ITextRenderer;
+
+import static org.thymeleaf.templatemode.TemplateMode.HTML;
+
+@Wid(widfile = "GeneratePDFDefinitions.wid", name = "GeneratePDF",
+        displayName = "GeneratePDF",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.pdf.GeneratePDFWorkitemHandler()",
+        documentation = "${artifactId}/index.html",
+        parameters = {
+                @WidParameter(name = "TemplateDocument", required = true),
+                @WidParameter(name = "TemplateData", required = true),
+                @WidParameter(name = "TemplateDataKey", required = true),
```

Comment:
Do we need the key? Why not setball work item parameters as context variables?

Reply:
Can you explain, sorry I dont understand

Reply:
what I meant is that in https://github.com/kiegroup/jbpm-work-items/pull/64/files#diff-483633a58c4867c278770e85cde2898aR96 you add the data based on the key into the context for template processing why not add all of workItem.getParameters() instead?
Imagine you generate an invoice so you have order, customer, seller etc so you will have many data items referenced in the template